ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -1947,30 +2024,70 @@ prop_diffusion_async_demotions defaultBearerInfo diffScript =
                            Just $ Right demotions
                          where
                            demotions = Set.singleton (remoteAddress connId)
+                       DiffusionPeerSelectionActionsTrace (PeerStatusChanged (WarmToCooling connId)) ->
+                           Just $ Right demotions
+                         where
+                           demotions = Set.singleton (remoteAddress connId)
+                       DiffusionConnectionManagerTrace (TrConnectionCleanup connId) ->
+                           Just $ Left failures
+                         where
+                           failures = Just $ Set.singleton (remoteAddress connId)
                        DiffusionPeerSelectionTrace (TraceDemoteAsynchronous status) ->
                            Just $ Left (Just failures)
                          where
-                           failures = Map.keysSet (Map.filter ((==PeerCold) . fst) status)
+                           failures = Map.keysSet (Map.filter ((==PeerCooling) . fst) status)
                        DiffusionPeerSelectionTrace (TraceDemoteBigLedgerPeersAsynchronous status) ->
                            Just $ Left (Just failures)
                          where
-                           failures = Map.keysSet (Map.filter ((==PeerCold) . fst) status)
+                           failures = Map.keysSet (Map.filter ((==PeerCooling) . fst) status)
                        DiffusionPeerSelectionTrace (TraceDemoteLocalAsynchronous status) ->
                            Just $ Left (Just failures)
                          where
-                           failures = Map.keysSet (Map.filter ((==PeerCold) . fst) status)
+                           failures = Map.keysSet (Map.filter ((==PeerCooling) . fst) status)
                        DiffusionPeerSelectionTrace (TraceDemoteHotFailed _ _ peeraddr _) ->
                            Just $ Left (Just failures)
                          where
                            failures = Set.singleton peeraddr
+                       DiffusionPeerSelectionTrace (TraceDemoteWarmDone _ _ addr) ->
+                           Just $ Left failures
+                         where
+                           failures = Just $ Set.singleton addr
+                       DiffusionPeerSelectionTrace (TraceDemoteWarmBigLedgerPeerDone _ _ addr) ->
+                           Just $ Left failures
+                         where
+                           failures = Just $ Set.singleton addr
+                       DiffusionPeerSelectionTrace (TraceDemoteHotDone _ _ addr) ->
+                           Just $ Left failures
+                         where
+                           failures = Just $ Set.singleton addr
+                       DiffusionPeerSelectionTrace (TraceDemoteHotBigLedgerPeerDone _ _ addr) ->
+                           Just $ Left failures
+                         where
+                           failures = Just $ Set.singleton addr
```

Comment:
Why `TraceDemote{Hot,Warm}{,BigLedgerPeer}Done` records `failures`, it should be fine to demote a ledger peer.

Reply:
You are right!